river/Root.zig

```diff
@@ -400,7 +400,10 @@ fn commitTransaction(self: *Self) void {
             view.dropSavedBuffers();
         }
 
-        if (view_tags_changed) output.sendViewTags();
+        if (view_tags_changed) {
+            output.sendViewTags();
+            output.sendUrgentTags();
+        }
```

Comment:
This logic isn't right, we need to be checking for the 3 cases I listed in another comment:

> 1. the urgent state of some view on the output changes
> 2. the tags of an urgent view change
> 3. an urgent view is sent to a different output (need to send new set of urgent tags for both outputs)

Only 1 and 2 are relevant here though and I see you've already handled 3. I'd suggest introducing an `urgent_tags_dirty` bool similar in usage to `view_tags_changed`.

Reply:
Should be better now, I'm not sure if 2 is implemented right though